kube-client/src/client/client_ext.rs

```diff
@@ -256,4 +395,60 @@ mod test {
 
         Ok(())
     }
+
+    #[tokio::test]
+    #[ignore = "needs cluster (will get svcs, clusterroles, pods, nodes)"]
+    async fn client_ext_fetch_ref_pods_svcs() -> Result<(), Box<dyn std::error::Error>> {
+        use k8s_openapi::api::{
+            core::v1::{Node, ObjectReference, Pod, Service},
+            rbac::v1::ClusterRole,
+        };
+
+        let client = Client::try_default().await?;
+        // namespaced fetch
+        let svc: Service = client
+            .fetch(&Ref::new(
+                ObjectReference {
+                    kind: Some(Service::kind(&()).into()),
+                    api_version: Some(Service::api_version(&()).into()),
+                    name: Some("kubernetes".into()),
+                    namespace: Some("default".into()),
+                    ..Default::default()
+                }
+                .into(),
+            ))
+            .await?;
```

Comment:
As a quick initial comment; I do think the interface here is promising and compelling. I also feel the system should ideally be able to just take a raw `ObjectReference` rather than have to `Ref`  wrap it (maybe it could work to take an `impl ObjectRef` in the signature, and do `impl ObjectRef for ObjectReference` to allow multiplexing between various k8s-openapi ObjectRef* types?).

Reply:
I actually tried to use existing `ObjectRef` initially, but faced with an import cycle trying to import the trait located in `kube-runtime`. Is this the one you refer to? It would be a great fit for this PR, and I agree. Maybe the `unstable-client` can be expanded first to `kube-runtime` and some bits live there, like automatic trait implementation, meanwhile the trait for referencing objects and used in fetch can be in the `client_ext`. This would simplify things

Reply:
ah, sorry, i actually forgot about `kube_runtime::ObjectRef` when I wrote the above comment. I was more considering creating a __new trait++ that would be able to convert from various `k8s_openapi` ObjectReference* style types (similar to what you have written, but with our own impls for the k8s-openapi types).

The `ObjectRef` we have in `kube_runtime` is a bit awkward and has a very particular use case therein and is not really relevant for the client work here atm.

Reply:
Just added similarly named one in the package. Seems to be working with `OwnerReference` and `ObjectReference`. Went with easier path to have `within` method for `OwnerReferences` for ns/dyn scoped resources, so there might be a “footgun” scenario when people try a random namespace outside the original resource, or a dynamic resource actually being a cluster scoped under the hood. But currently version looks ok to me, WDYT?

Reply:
So… After some testing I reworked the patch to do stuff with `DynamicObject`. This is the main use-case I have in mind, where essentially an equivalent of `unstructured` is taken as is, and then it is attempted to be deserialized into actual object with some business logic assigned to it. Sorry if this does not look good, trying my best… If there is a better way I’d be glad to use it.